### ISSUE TITLE
Unidentified error with azure authentication in Azure Data Studio

### ISSUE THREAD

**Issue by kn6405** [microsoft/azuredatastudio#15532]
Issue Type: <b>Bug</b>

When trying to "Sign in to Azure..." it takes me to the login page, calls my phone, and then the browser page displays "Unidentified error with azure authentication". 

Azure Data Studio version: azuredatastudio 1.29.0 (0f5cfdc2c8c2433028842e16dc5ac1a65da24292, 2021-05-15T01:12:11.112Z)
OS version: Windows_NT x64 10.0.18363

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-10810U CPU @ 1.10GHz (12 x 1608)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.78GB (17.47GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (1)</summary>

Extension|Author (truncated)|Version
---|---|---
azuredatastudio-postgresql|Mic|0.2.7

</details>

**Comment by kburtram** [comment]
@kn6405 the authentication in the web browser is completely decoupled from Azure Data Studio application code.  I'm not sure how best to investigate but likely there issue is happening in the Azure AAD service (if I had to guess).  If possible, it may be helpful to check if AAD is working fine from other applications, such as SSMS or VSCode?

@cssuh do you have any suggestions how to investigate?

**Comment by andywperry** [comment]
I have been running into the same problem for over a year.  I can log into our Azure SQL Server using AAD from SSMS.  But when I use the exact same credentials in Azure Data Studio, I receive an error, sometimes referencing a missing token and sometimes just "connection failed."

Note that the OP referenced being taken to a login page.  For Azure Data Studio, I am taken to a web page to verify my credentials (which seems to work--there's no error at that stage, only after I hit the "connect" button inside the app).  For SSMS, a separate dialog pops up inside SSMS without going through a web browser.  That's the only difference I can see between the successful connection and the failed one.

**Comment by kn6405** [comment]
> @kn6405 the authentication in the web browser is completely decoupled from Azure Data Studio application code. I'm not sure how best to investigate but likely there issue is happening in the Azure AAD service (if I had to guess). If possible, it may be helpful to check if AAD is working fine from other applications, such as SSMS or VSCode?
> 
> @cssuh do you have any suggestions how to investigate?

Yah, not sure. I have AAD working for azure cli, azure storage explorer, and visual studio code. With those it takes me to the same looking azure login webpage, I successfully login, and then the webpage says it can be closed and I'm good to go. So I'm not sure why in Azure Data Studio after the login page and successfully entering my credentials it goes to the error webpage that is just a white page with the text "Unidentified error with azure authentication in Azure Data Studio". I checked the logs for azure data studio and didn't notice anything that would help pinpoint or identify the isssue. And considering it working in other apps I'm not sure why its just isolated to this one app so far.

**Comment by cssuh** [comment]
Hi @kn6405, do you see any error messages in the Developer Tools Console?

**Comment by thec0dewriter** [comment]
Hello,
I have the same error and in the dev console appears the following error
`log.ts:197   ERR Cannot read property 'accountType' of undefined: TypeError: Cannot read property 'accountType' of undefined
    at Po.renderElement (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:877:62864)
    at Po.renderElement (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:877:63518)
    at H.renderElement (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:286:1590)
    at f.insertItemInDOM (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:272:17275)
    at f._splice (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:272:13081)
    at f.splice (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:272:11295)
    at file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:29:13406
    at Array.forEach (<anonymous>)
    at t.splice (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:29:13393)
    at file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:286:7593
    at p.bufferEvents (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:85:3688)
    at G.splice (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:286:7560)
    at F.updateAccounts (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:3255:30852)
    at V.updateProviderAccounts (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:3255:36448)
    at file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:3255:32295
    at a.fire (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:85:1835)
    at file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:875:10660
    at a.fire (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:85:1835)
    at pc.fireAccountListUpdate (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:3255:44036)
    at file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:3255:40109
    at async Rr.run (file:///C:/Program Files/Azure Data Studio/resources/app/out/vs/workbench/workbench.desktop.main.js:877:18821)
console.ts:137 [Extension Host] Shutting off webserver...
`

**Comment by hannesroth88** [comment]
I also get 
"ERR Cannot read property 'accountType' of undefined"
has this been solved?
I am able to login using VSCODE with Azure

**Comment by cssuh** [comment]
Closing as there has been no activity on this issue in a few months. Please feel free to open a new issue if you are still having problems.
